packages/backend-tasks/src/tasks/reader.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { HumanDuration, TaskScheduleDefinition } from './types';
+import { Duration } from 'luxon';
+
+function readDuration(config: Config, key: string): Duration | HumanDuration {
+  return typeof config.get(key) === 'string'
+    ? Duration.fromISO(config.getString(key))
+    : (config.getConfig(key) as HumanDuration);
+}
+
+function readCronOrDuration(
+  config: Config,
+  key: string,
+): { cron: string } | Duration | HumanDuration {
+  const value = config.get(key);
+  if (typeof value === 'object' && (value as { cron?: string }).cron) {
+    return value as { cron: string };
+  }
+
+  return readDuration(config, key);
+}
+
+/**
+ * Reads a TaskScheduleDefinition from a Config.
+ * Expects the config not to be the root config,
+ * but the config for the definition.
+ *
+ * @param config - config for a TaskScheduleDefinition.
+ * @public
+ */
+export function readTaskScheduleDefinition(
```

Comment:
missing some tests for these functions :)

Reply:
sure, I will add for all new stuff and thanks for the pointer above 👍